Fix congrats footer detail click tracking

chad1008 opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to 5794-gh-Automattic/dotcom-forge

## Proposed Changes

* Update the Tracks events for the Domain-only congrats page
* Update the Tracks events for the Plan-only congrats page
* Add footer tracking to the Titan congrats page
* Per [feedback](https://github.com/Automattic/wp-calypso/pull/87912#discussion_r1503885524), add a `context` event prop to all footer links

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Purchase a domain
* On the congrats page, confirm that clicking the two footer links fires a `calypso_thank_you_footer_link_click` tracks event
  * The link under **Dive into domain essentials** should carry a `type` prop of `footer-domain-essentials`
  * The link under **Domain support resources** should carry a `type` prop of `footer-domain-resources`
  * Both links should carry a `context` prop of `domain-only`
* Purchase a plan
* On the congrats page, confirm that clicking the two footer links fires a `calypso_thank_you_footer_link_click` tracks event
  * The link under **A site refresh** should carry a `type` prop of `footer-site-refresh`
  * The link under **Everything you need to know** should carry a `type` prop of `footer-support`
  * Both links should carry a `context` prop of `plan-only`
* Purchase Professional Email
* On the congrats page, confirm that clicking the two footer links fires a `calypso_thank_you_footer_link_click` tracks event
  * The link under **Manage your email and site from anywhere** should carry a `type` prop of `footer-manage-email`
  * The link under **Email questions? We have the answers** should carry a `type` prop of `footer-questions-email`
  * Both links should carry a `context` prop of `titan-setup`

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

chad1008 committed:
update plan and domain congrats tracking

chad1008 committed:
add tracks to titan congrats page

fredrikekelund commented on client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/content/get-domain-footer-details.ts:
```diff
@@ -25,7 +27,9 @@ export default function getDomainFooterDetails( limit?: number ): ThankYouFooter
 			buttonOnClick: () => {
-				recordTracksEvent( 'calypso_thank_you_footer-domain-resources' );
+				recordTracksEvent( 'calypso_thank_you_footer_link_click', {
+					type: 'footer-domain-resources',
```
```suggestion
					type: 'domain-resources',
```

fredrikekelund commented on client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/plan-only.tsx:
```diff
@@ -42,7 +42,7 @@ export default function PlanOnlyThankYou( { primaryPurchase }: PlanOnlyThankYouP
 			buttonHref: `/people/new/${ siteSlug }`,
 			buttonOnClick: () => {
-				recordTracksEvent( 'calypso_plan_thank_you_add_members_click' );
+				recordTracksEvent( 'calypso_thank_you_footer_link_click', { type: 'footer-add-members' } );
```
```suggestion
				recordTracksEvent( 'calypso_thank_you_footer_link_click', { type: 'add-members' } );
```

fredrikekelund commented on client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/plan-only.tsx:
```diff
@@ -56,7 +56,7 @@ export default function PlanOnlyThankYou( { primaryPurchase }: PlanOnlyThankYouP
 			buttonHref: `/themes/${ siteSlug }`,
 			buttonOnClick: () => {
-				recordTracksEvent( 'calypso_plan_thank_you_theme_click' );
+				recordTracksEvent( 'calypso_thank_you_footer_link_click', { type: 'footer-site-refresh' } );
```
```suggestion
				recordTracksEvent( 'calypso_thank_you_footer_link_click', { type: 'site-refresh' } );
```

fredrikekelund commented on client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/plan-only.tsx:
```diff
@@ -68,7 +68,7 @@ export default function PlanOnlyThankYou( { primaryPurchase }: PlanOnlyThankYouP
 		buttonHref: '/support',
 		buttonOnClick: () => {
-			recordTracksEvent( 'calypso_plan_thank_you_support_click' );
+			recordTracksEvent( 'calypso_thank_you_footer_link_click', { type: 'footer-support' } );
```
```suggestion
			recordTracksEvent( 'calypso_thank_you_footer_link_click', { type: 'support' } );
```

fredrikekelund commented on client/my-sites/email/titan-set-up-thank-you/index.tsx:
```diff
@@ -55,16 +56,24 @@ const TitanSetUpThankYou = ( {
 			buttonHref: '/support/category/domains-and-email/email/',
+			buttonOnClick: () => {
+				recordTracksEvent( 'calypso_thank_you_footer_link_click', {
+					type: 'footer-questions-email',
```
```suggestion
					type: 'questions-email',
```

fredrikekelund commented on client/my-sites/email/titan-set-up-thank-you/index.tsx:
```diff
@@ -55,16 +56,24 @@ const TitanSetUpThankYou = ( {
 					context: 'checkout-thank-you',
 				} );
+				recordTracksEvent( 'calypso_thank_you_footer_link_click', {
+					type: 'footer-manage-email',
```
```suggestion
					type: 'manage-email',
```

fredrikekelund commented on client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/content/get-domain-footer-details.ts:
```diff
@@ -13,7 +13,9 @@ export default function getDomainFooterDetails( limit?: number ): ThankYouFooter
 			buttonHref: '/support/domains',
 			buttonOnClick: () => {
-				recordTracksEvent( 'calypso_thank_you_footer-domain-essentials' );
+				recordTracksEvent( 'calypso_thank_you_footer_link_click', {
+					type: 'footer-domain-essentials',
+				} );
```
```suggestion
				recordTracksEvent( 'calypso_thank_you_footer_link_click', {
					context: context, // 'domain-only' or 'transfer-to-user'
					type: 'domain-essentials',
				} );
```
Since we already have `footer` in the event name, I think mentioning it in the type prop might be superfluous.

I wonder if we shouldn't pass a `context` prop with this event as well. While it's easy enough to keep track of the fact that `domain-essentials` belongs to the domain congrats page, I'm not sure this will hold true for all footer links on all congrats pages.

Here in `getDomainFooterDetails`, which is used in both `DomainTransferToAnyUser` and `DomainOnlyThankYou`, we should consider using a new argument passed to `getDomainFooterDetails` for the `context` prop.

Side note: is there a convention around dashes vs underscores for Tracks event properties..?

wongasy replied:
>I wonder if we shouldn't pass a `context` prop with this event as well. 

Indeed the same footer blurb may be shown in different Congrats pages. It'd be useful to pass a `context` prop to identify where the link came from. e.g. `plan-only`, `domain-only`, etc. Although, I'd suggest going live with what we have in this PR so we get the typo fixed asap. Alternatively, create a separate PR that fixes the typo, get that deployed, then rebase this branch.

>is there a convention around dashes vs underscores for Tracks event properties..?

From what I've seen, tracks event names and property _names_ use `_`. I'm not aware of conventions for property _values_, but here's an example that uses `-` for reporting property values:
https://github.com/Automattic/wp-calypso/blob/fbbd52ec657736dfb967e6d29b6f493bcc4d0e36/apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/notices.js#L24-L27

fredrikekelund reviewed (commented):
Looking good overall! Left a few comments for your consideration

chad1008 commented:
New PR to get that typo fixed up more speedily: #87952

chad1008 committed:
remove redundancy

Co-authored-by: Fredrik Rombach Ekelund <fredrik@fredrik.computer>

chad1008 committed:
remove redundancy

Co-authored-by: Fredrik Rombach Ekelund <fredrik@fredrik.computer>

chad1008 committed:
remove redundancy

Co-authored-by: Fredrik Rombach Ekelund <fredrik@fredrik.computer>

chad1008 committed:
remove redundancy

Co-authored-by: Fredrik Rombach Ekelund <fredrik@fredrik.computer>

chad1008 committed:
remove redundancy

Co-authored-by: Fredrik Rombach Ekelund <fredrik@fredrik.computer>

chad1008 committed:
remove redundancy

Co-authored-by: Fredrik Rombach Ekelund <fredrik@fredrik.computer>

chad1008 committed:
add `context` event prop to plans and titan

chad1008 committed:
add `context` arg and event prop for domain congrats footers

chad1008 commented:
Thank you both @wongasy and @fredrikekelund!

I've added the new `context` prop, cleaned up the repetitive `footer` naming, and updated the PR description/testing steps accordingly. PR has also been rebased to include the typo fix that was addressed separately. 🙂

fredrikekelund commented on client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/content/get-domain-footer-details.ts:
```diff
@@ -13,7 +16,10 @@ export default function getDomainFooterDetails( limit?: number ): ThankYouFooter
-				recordTracksEvent( 'calypso_thank_you_footer-domain-essentials' );
+				recordTracksEvent( 'calypso_thank_you_footer_link_click', {
+					context: context,
+					type: 'footer-domain-essentials',
```
```suggestion
					type: 'domain-essentials',
```
One last fix!

fredrikekelund reviewed (approved):
Looks like there's still one `type` prop with `footer` in the name. Other than that, though, this LGTM 👍

wongasy commented on client/my-sites/email/titan-set-up-thank-you/index.tsx:
```diff
@@ -55,6 +56,10 @@ const TitanSetUpThankYou = ( {
 				} );
+				recordTracksEvent( 'calypso_thank_you_footer_link_click', {
+					context: 'titan-setup',
+					type: 'manage-email',
```
```suggestion
					type: 'get-the-app',
```
Let's change the key and the event prop to `get-the-app` instead, as `manage-email` seems to suggest the link takes users to the Email Management Page, but that isn't the case.

wongasy reviewed (approved):
LGTM and tests well! Once the two final feedbacks are addressed, we're good to go!

Click tracking for footer links works as expected for the following pages:

#### Domain transfer congrats page (prop to be updated per https://github.com/Automattic/wp-calypso/pull/87912#discussion_r1505728956)

<img width="423" alt="Screen Shot 2024-02-28 at 8 24 04 AM" src="https://github.com/Automattic/wp-calypso/assets/104910361/e067fd20-6ada-45be-ac28-60bb0199d3be">

<img width="469" alt="Screen Shot 2024-02-28 at 8 23 54 AM" src="https://github.com/Automattic/wp-calypso/assets/104910361/37a3c846-776c-4741-8c42-8880786eebe8">

#### Titan congrats page (prop to be updated per https://github.com/Automattic/wp-calypso/pull/87912#discussion_r1505957486)

<img width="437" alt="Screen Shot 2024-02-28 at 8 23 33 AM" src="https://github.com/Automattic/wp-calypso/assets/104910361/04fbee05-3d68-4214-a962-2e8cd4d03ae8">

<img width="444" alt="Screen Shot 2024-02-28 at 8 17 21 AM" src="https://github.com/Automattic/wp-calypso/assets/104910361/1582166c-f0a0-404e-85b3-d063c83b1dea">

#### Plan congrats page (prop to be updated per feedback)

<img width="456" alt="Screen Shot 2024-02-28 at 8 14 17 AM" src="https://github.com/Automattic/wp-calypso/assets/104910361/9d397f07-f791-464a-b7fe-04d84900d8a3">

<img width="428" alt="Screen Shot 2024-02-28 at 8 13 58 AM" src="https://github.com/Automattic/wp-calypso/assets/104910361/6155c691-35a6-4bcb-8950-168b9304aea7">

chad1008 committed:
implement feedback

Co-authored-by: Fredrik Rombach Ekelund <fredrik@fredrik.computer>

chad1008 committed:
rename `manage-email` to `get-the-app`
